src/shogun/machine/gp/SingleSparseInferenceBase.cpp

```diff
@@ -29,23 +29,81 @@
  *
  */
 
-
 #include <shogun/machine/gp/SingleSparseInferenceBase.h>
 #ifdef USE_GPL_SHOGUN
 #ifdef HAVE_NLOPT
-#include <nlopt.h>
+#include <shogun/optimization/NLOPTMinimizer.h>
 #include <shogun/features/DenseFeatures.h>
 #endif //HAVE_NLOPT
 #endif //USE_GPL_SHOGUN
 
-
 #include <shogun/mathematics/Math.h>
 #include <shogun/mathematics/eigen3.h>
 #include <shogun/features/DotFeatures.h>
+#include <shogun/optimization/FirstOrderBoundConstraintsCostFunction.h>
 
 using namespace shogun;
 using namespace Eigen;
 
+namespace shogun
+{
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+/** Wrapper cost function used for the NLOPT minimizer */
+class CSingleSparseInferenceCostFunction: public FirstOrderBoundConstraintsCostFunction
+{
+public:
+        CSingleSparseInferenceCostFunction():FirstOrderBoundConstraintsCostFunction() {  init(); }
+        virtual ~CSingleSparseInferenceCostFunction() { SG_UNREF(m_obj); }
+        void set_target(CSingleSparseInferenceBase *obj)
+	{
+		if(obj!=m_obj)
+		{
+			SG_REF(obj);
+			SG_UNREF(m_obj);
+			m_obj=obj;
+			REQUIRE(m_obj,"Object not set\n");
+			m_obj->check_fully_sparse();
+			REQUIRE(m_obj->m_fully_sparse,"Can not compute gradient\n");
```

Comment:
Proper error message. State why it cannot do that, what is given and what is expected